src/pngchunk_int.cpp

```diff
@@ -162,6 +164,9 @@ namespace Exiv2 {
         }
         else if(type == iTXt_Chunk)
         {
+            const int nullSeparators = std::count(&data.pData_[keysize+3], &data.pData_[data.size_-1], '\0');
```

Comment:
Uh, I think this should be `&data.pData_[data.size_]` as the end iterator should point to the first element **outside** of the range.

Reply:
ahg, you are right. I'll create a mini PR fixing that.